### DISCUSSION TITLE
Display no. of lines of code

### DISCUSSION THREAD

**Discussion by VinayakChanekar** [pmd/pmd#3924] [Q&A] [answered]
Is there any feature/config that can let the tool display the number of lines of code without the comments?

**Comment by oowekyala** [comment, marked as answer]
The java module has a rule [NcssCount](https://pmd.github.io/latest/pmd_rules_java_design.html#ncsscount), and the apex module has eg [NcssTypeCount](https://pmd.github.io/latest/pmd_rules_apex_design.html#ncsstypecount) and a couple others. These rules allow you to enforce that some code constructs stay under a certain limit of Ncss (non-commenting source statements). As of today, PMD has no built-in facility to just report metrics. As a workaround you could set the rule reporting threshold to 1 and get one violation per class and the Ncss value in the message.

If this does not answer your question please add more details about how you want to access the results
